### PR TITLE
[RET-142] Textfield Validation

### PR DESCRIPTION
**Pull request by jcmagsay**
### What's this PR do?
Validation for: email & zip
Create mask for: Password
Add error and focus states for textfields

### What are the relevant tickets?
https://mondorobot.atlassian.net/browse/RET-142

### Screenshots (if appropriate)
screen shot 2017-04-21 at 11 12 39 am

### Who should review this PR?
@jcmagsay @mhfen

### CODE REVIEW

File: src/client/components/textfield.jsx [hunk 1]
```diff
@@ -8,8 +8,10 @@ export default class Textfield extends React.Component {
     this.inputVal;
 
     this.state = {
+      'isDirty': false,
+      'isError': false,
       'isFocused': false,
-      'isDirty': false
+      'fieldValid': true
```
Comment (mhfen) [inline src/client/components/textfield.jsx]: I like this use of state and the clearly defined ways an input can be given state.

File: src/client/components/textfield.jsx [hunk 2]
```diff
@@ -44,11 +74,81 @@ export default class Textfield extends React.Component {
    *
    * @method  blur
    */
-  blur() {
-    this.setState({
-      'isFocused': false,
-      'isDirty': this.inputVal
-    });
+  blur(event) {
```
Comment (mhfen) [inline src/client/components/textfield.jsx]: Is there a way to check and update these states when the form is submitted without `blur`, like when hitting `ENTER` while still focused on an input?

Comment (jcmagsay) [inline src/client/components/textfield.jsx]: I like that idea a lot. We can definitely add this handler, but I need a working example like in an Authentication page. How about I add this the to acceptance criterial for the submission of the Login page form? That way we can truly test the functionality

Comment (mhfen) [inline src/client/components/textfield.jsx]: Sounds good to me

**Pull request merged**